src/App/Pages/Settings/SettingsPage/SettingsPageViewModel.cs

```diff
@@ -130,9 +130,12 @@ public async Task InitAsync()
             _vaultTimeoutDisplayValue = _vaultTimeoutOptions.FirstOrDefault(o => o.Value == _vaultTimeout).Key;
             _vaultTimeoutDisplayValue ??= _vaultTimeoutOptions.Where(o => o.Value == CustomVaultTimeoutValue).First().Key;
 
-            var savedVaultTimeoutAction = await _vaultTimeoutService.GetVaultTimeoutAction();
-            var timeoutAction = savedVaultTimeoutAction ?? VaultTimeoutAction.Lock;
-            if (!_hasMasterPassword && timeoutAction == VaultTimeoutAction.Lock)
+
+            var pinSet = await _vaultTimeoutService.GetPinLockTypeAsync();
+            _pin = pinSet != PinLockType.Disabled;
+            _biometric = await _vaultTimeoutService.IsBiometricLockSetAsync();
+            var timeoutAction = await _vaultTimeoutService.GetVaultTimeoutAction() ?? VaultTimeoutAction.Lock;
+            if (!IsVaultTimeoutActionLockAllowed)
```

Comment:
🤔 I see that before it was also being considered the timeout action to be `Lock`, is that not needed anymore? Because if action is `Logout` then inside the if, it's going to reassign `Logout` and set it in the service, which I don't think it would be necessary.

Reply:
Yup you are right we can check for Lock to run the code inside and avoid unnecessary SetVaultTimeout